store/streaming/file/service.go

```diff
@@ -288,3 +178,31 @@ func isDirWriteable(dir string) error {
 	}
 	return os.Remove(f)
 }
+
+func writeLengthPrefixedFile(path string, data []byte, fsync bool) (err error) {
+	var f *os.File
+	f, err = os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o600)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := f.Close(); err1 != nil && err == nil {
```

Comment:
I can't see why we need to reference `err` here? All error cases are handled in the main function.

Reply:
This pattern is to avoid last err overwritten by close err.

Reply:
BTW, the same pattern in golang stdlib: https://cs.opensource.google/go/go/+/refs/tags/go1.19.3:src/os/file.go;l=722